vnext/ReactUWP/Modules/Animated/PropsAnimatedNode.cpp

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "PropsAnimatedNode.h"
+#include <ReactUWP\Modules\NativeUIManager.h>
+#include <Views/ShadowNodeBase.h>
+#include "StyleAnimatedNode.h"
+
+namespace react { namespace uwp {
+  PropsAnimatedNode::PropsAnimatedNode(int64_t tag, const folly::dynamic& config, const std::weak_ptr<IReactInstance>& instance, const std::shared_ptr<NativeAnimatedNodeManager>& manager) :
+    AnimatedNode(tag), m_instance(instance), m_manager(manager)
+  {
+    for (auto entry : config.find("props").dereference().second.items())
+    {
+      m_propMapping.insert({ entry.first.getString(), entry.second.getInt() });
+    }
+  }
+
+  void PropsAnimatedNode::ConnectToView(int64_t viewTag)
+  {
+    if (m_connectedViewTag != s_connectedViewTagUnset)
+    {
+      throw new std::invalid_argument("Animated node " + std::to_string(m_tag) + " has already been attached to a view already exists.");
+      return;
+    }
+    m_connectedViewTag = viewTag;
+    UpdateView();
+  }
+
+  void PropsAnimatedNode::DisconnectFromView(int64_t viewTag)
+  {
+    if (m_connectedViewTag != viewTag)
+    {
+      throw new std::invalid_argument("Attempting to disconnect view that has not been connected with the given animated node.");
+      return;
+    }
+
+    std::vector<int64_t> keys{};
+    for (auto anim : m_expressionAnimations)
+    {
+      keys.push_back(anim.first);
+    }
+    for(auto key : keys)
+    {
+      DisposeCompletedAnimation(key);
+    }
+
+    if (m_centerPointAnimation)
+    {
+      if (auto target = GetUIElement())
+      {
+        target.StopAnimation(m_centerPointAnimation);
+      }
+      m_centerPointAnimation = nullptr;
+    }
+
+    m_connectedViewTag = s_connectedViewTagUnset;
+    m_needsCenterPointAnimation = false;
+  }
+
+  void PropsAnimatedNode::RestoreDefaultValues()
+  {
+
+  }
+
+  void PropsAnimatedNode::UpdateView()
+  {
+    if (m_connectedViewTag == s_connectedViewTagUnset)
+    {
+      return;
+    }
+
+    if (const auto manager = m_manager.lock())
+    {
+      for (auto entry : m_propMapping)
+      {
+        if (manager->m_styleNodes.count(entry.second))
+        {
+          for (auto styleEntry : manager->m_styleNodes.at(entry.second)->GetMapping())
+          {
+            MakeAnimation(styleEntry.second, styleEntry.first);
+          }
+        }
+        else if (manager->m_valueNodes.count(entry.second))
+        {
+          MakeAnimation(entry.second, StringToFacadeType(entry.first));
+        }
+      }
+    }
+
+    StartAnimations();
+  }
+
+  void PropsAnimatedNode::StartAnimations()
+  {
+    if (m_expressionAnimations.size())
+    {
+      if (const auto uiElement = GetUIElement())
+      {
+        uiElement.RotationAxis(m_rotationAxis);
+        for (const auto anim : m_expressionAnimations)
```

Comment:
const auto&

Reply:
missed this one?

---
In reply to: [297483713](https://github.com/microsoft/react-native-windows/pull/2556#discussion_r297483713) [](ancestors = 297483713)